functions/wav_time.h

```diff
@@ -0,0 +1,143 @@
+#ifndef FUNCTIONS_WAV_TIME_H
+#define FUNCTIONS_WAV_TIME_H
+
+#include "wav_effect.h"
+#include "int.h"
+#include "../blades/blade_base.h"
+
+// Usage: WavLen<EFFECT_NAME>  
+// EFFECT_NAMENUM : any off the available effect form the style edittor
+// Returns length of wav file returned as const int in [ms]
+// returned value: INTEGER
+//
+template<BladeEffectType EFFECT>
+class WavLen {
+ public:
+  void run(BladeBase* blade) {
+    found_ = false;
+    eff_name_ = wav_effect.GetEffectName(EFFECT);//find correspondig SoundEffectName for BladeEffectType
+    for (size_t i = 0; i < NELEM(wav_players); i++) {
+        if(strstr(eff_name_, wav_players[i].effectname()) && wav_players[i].isPlaying()) {found_ = true; idx=(int)i;}
+        if(strcmp(eff_name_,"pstoff")==0) {if(strstr(eff_name_, wav_players[i].effect2ndname()) && wav_players[i].isPlaying()) {found_ = true; idx=(int)i;}}
+            if (found_) { value_ = clampi32(wav_players[idx].length()*1000, 0, 32768);}
+            if (found_ && !shown_) {
+                STDOUT << "WavLen: eff_name:'" << eff_name_  << "', effectname: '";
+                if(strcmp(eff_name_,"pstoff")==0) { STDOUT << wav_players[i].effect2ndname();} else { STDOUT << wav_players[i].effectname();}
+                STDOUT << "' wav_player:" << i 
+                       << ", filename=" << wav_players[i].filename() 
+                       << ", length: " << wav_players[i].length() << "[s]," 
+                       << " converted = " << value_ << "[ms]\n";
+            shown_ = true;
+            }
+    }
+    if (!found_){
+        if(shown_) {STDOUT << "WavLen stopped for effect '" << eff_name_ <<"', value retained is: " << value_ << "\n";}
+        //should we retain value after the effect has gone? Only reset it when new effect is selected?
+        //value_ = 32767;
+        shown_ = false;
+    }
+  }
+  int getInteger(int led) { return value_; }
```

Comment:
In some cases (transitions for example) ProffieOS tends to snapshot the length of the transition when the transitions start. I don't see how this code can guarantee that the length of the associated wav file is actually known at that time.

Reply:
Seems to be working fine, even for styles that have multiple transitions in them.